veniq/dataset_mining/git_api.py

```diff
@@ -0,0 +1,271 @@
+    return files_after_arr
+
+
+def find_filename(filename_raw: Path, files):
+    return [x for x in files if x['filename'].find(filename_raw.as_posix()) > -1]
+
+
+def add_to_dict_set(key, val, multi_dict):
+    if key not in multi_dict:
+        multi_dict[key] = set()
+    temp_set = multi_dict[key]
+    temp_set.add(val)
+    multi_dict[key] = temp_set
+
+
+def filter_refactorings_by_em(dataset_samples):
+    handled_samples = []
+    for sample in dataset_samples:
+        for x in sample['refactorings']:
+            if x['type'] == 'Extract Method':
+                new_item = sample.copy()
+                new_item['refactorings'] = x
+                handled_samples.append(new_item)
+
+    return handled_samples
+
+
+if __name__ == '__main__':
+    system_cores_qty = os.cpu_count() or 1
+    parser = ArgumentParser()
+    parser.add_argument(
+        "-o", "--output_dir",
+        help="Path where commits with files will be saved",
+        required=True
+    )
+    parser.add_argument(
+        "-t", "--token",
+        help="Token for github API",
+        required=True
+    )
+    parser.add_argument(
+        "--jobs",
+        "-j",
+        type=int,
+        default=system_cores_qty - 1,
+        help="Number of processes to spawn. "
+             "By default one less than number of cores. "
+             "Be careful to raise it above, machine may stop responding while creating dataset.",
+    )
+    parser.add_argument(
+        "-s", "--dataset_file",
+        help="Json with dataset",
+        required=True,
+        type=str,
+    )
+    args = parser.parse_args()
+    output_dir = Path(args.output_dir)
+
+    if not output_dir.exists():
+        output_dir.mkdir(parents=True)
+
+    output_df = pd.DataFrame(
+        columns=[
+            'id',
+            'repo_url',
+            'class_name',
+            'file_name_after',
+            'file_name_before',
+            'downloaded_after',
+            'downloaded_before',
+            'commit_before',
+            'commit_after',
+        ]
+    )
+    with open(args.dataset_file, encoding='utf-8') as f:
+        with Manager() as manager:
+            d = manager.dict()  # type: ignore
+            with ProcessPool(system_cores_qty) as executor:
+                dataset_samples = json.loads(f.read())
+
+                handled_samples = filter_refactorings_by_em(dataset_samples)
+
+                func = partial(handle_commit_example, output_dir=output_dir, token=args.token, classes_dict=d)
+                future = executor.map(func, handled_samples, timeout=10000, )
+                result = future.result()
+
+                for sample in tqdm(handled_samples, total=len(handled_samples)):
+                    try:
+                        results = next(result)
+                        for res in results:
+                            output_df = output_df.append(res, ignore_index=True)
+                        output_df.to_csv('new_urls.csv')
```

Comment:
If I understand correctly, you here accumulate data in data frame and dump it for every sample. Why don't you dump it in the end? Is it for storing intermediate results? It can slow down your script.

Reply:
it doesn't matter. The speed is not needed here. This script is very slow and I need to get intermediate results